File: language-support/js/daml-ledger/src/mapping/get_transaction_by_event_id_request.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as mapping from '.';
import * as ledger from '..';
import * as grpc from 'daml-grpc';

export const GetTransactionByEventIdRequest: mapping.Mapping<grpc.GetTransactionByEventIdRequest, ledger.GetTransactionByEventIdRequest> = {
    toObject(message: grpc.GetTransactionByEventIdRequest): ledger.GetTransactionByEventIdRequest {
        return {
            eventId: message.getEventId(),
            requestingParties: message.getRequestingPartiesList()
        };
    },
    toMessage(object: ledger.GetTransactionByEventIdRequest): grpc.GetTransactionByEventIdRequest {
        const message = new grpc.GetTransactionByEventIdRequest();
        message.setEventId(object.eventId);
        message.setRequestingPartiesList(object.requestingParties);
        return message;
    }
}

File: sdk/canton/community/bindings-java/src/main/java/com/daml/ledger/javaapi/data/GetTransactionByEventIdRequest.java
// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates.
// Proprietary code. All rights reserved.

package com.daml.ledger.javaapi.data;

import com.daml.ledger.api.v2.UpdateServiceOuterClass;
import java.util.List;
import java.util.Objects;
import org.checkerframework.checker.nullness.qual.NonNull;

public final class GetTransactionByEventIdRequest {

  @NonNull private final String eventId;

  @NonNull private final List<@NonNull String> requestingParties;

  public GetTransactionByEventIdRequest(
      @NonNull String eventId, @NonNull List<@NonNull String> requestingParties) {
    this.eventId = eventId;
    this.requestingParties = List.copyOf(requestingParties);
  }

  @NonNull
  public String getEventId() {
    return eventId;
  }

  @NonNull
  public List<@NonNull String> getRequestingParties() {
    return requestingParties;
  }

  public static GetTransactionByEventIdRequest fromProto(
      UpdateServiceOuterClass.GetTransactionByEventIdRequest request) {
    return new GetTransactionByEventIdRequest(
        request.getEventId(), request.getRequestingPartiesList());
  }

  public UpdateServiceOuterClass.GetTransactionByEventIdRequest toProto() {
    return UpdateServiceOuterClass.GetTransactionByEventIdRequest.newBuilder()
        .setEventId(eventId)
        .addAllRequestingParties(requestingParties)
        .build();
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    GetTransactionByEventIdRequest that = (GetTransactionByEventIdRequest) o;
    return Objects.equals(eventId, that.eventId)
        && Objects.equals(requestingParties, that.requestingParties);
  }

  @Override
  public int hashCode() {

    return Objects.hash(eventId, requestingParties);
  }

  @Override
  public String toString() {
    return "GetTransactionByEventIdRequest{"
        + "eventId="
        + eventId
        + ", requestingParties="
        + requestingParties
        + '}';
  }
}
